OKTA Authentication swap to Auth0

da-vazquez opened the pull request:
# Pull Request Template

Replaced Okta embedded Login Form with a button component that routes the user to the Auth0 Universal Login page. Modified Api/index.js by removing authState (okta infrastructure) and replaced it with an idToken and Authentication boolean value pulled from localStorage that is then sent to the backend API for verification.

    1. User can login with their current email or password or sign-up on the same form (provided by Auth0)
    2. Auth0 form will be styled at a later time to match current SS look and feel
    
## Description
Motivation/purpose for the feature:
     1. To ensure the user can login / register and to replace the current Okta Authentication system currently in place. All of the features for SS require an authenticated user so this feature is critical to the user experience.
     2. Allows current architecture to remain functional using Auth0
 
- What were those changes
     1. A button was created on the LandingPage.js component that will direct the user to the Auth0 Login screen and re-direct back to the SS application after the credentials are verified.
     2. After the user logs in, the application stores the user's idToken and authenication status in localStorage so it can be consumed by the rest of the components
 
- What issues were fixed
     1. The embedded Okta sign-in widget was removed and replaced by a login button that directs to the Auth0 Login Page.
     2. Removed all instances of Okta on login component and parent dashboard component
     3. API/index.js removed all instances of authState and updated getAuthHeader to send idToken to API
     3. Tests updated to mock Auth0 instead of Okta
     4. Remaining components have also been modified on separate PR's and will be merged in multiple stages to ensure there are no bugs

 - Provide the link to the user story it is referencing in Trello board or other supporting document
     https://trello.com/c/uoBFFr9W/26-as-a-user-i-want-to-be-able-to-login-using-auth0-on-the-story-squad-website

## Notes

Additional modifications were made to ParentDashBoardContainer.js, ParentNavTopBar.js, and SecureRoute.js as they pertained to testing the Auth0 redirect functionality. Without the modifications, the application was crashing. Backend API Pull Request will be merged simultaneously with this PR.

## Commits Checklist
- [x] commits have clear title names 
- [x] commit has descriptive message of what has changed 
- [x] more small commits than less big commits 

## Code Checklist 
- [x] code is formatted appropriately 
- [x] code meets DRY standards
- [X] no dead code (check for logs and print statements) (a few commented out lines, will need additional clarification if code is needed or not)
- [x] no TODOS instead necessary comments 
- [x] code follows Story Squad standards for:
     - Language 
     - Framework 
     - Libraries   

### PRs needs to be reviewed by at least 2 team members
 - TPL is the 3rd reviewer 

[Pull Request Rubric](https://www.notion.so/1fc04e4fedeb429ba873b7c68d281707?v=74054da7991341c0bf970f39410c43da)

da-vazquez committed:
auth0 installed to replace current okta auth

da-vazquez committed:
wrapped App with auth0 provider to pass auth status and user info to store. replaced Okta secure routes to Auth0 secure routes

da-vazquez committed:
existing okta login widget replaced with button that directs to auth0 universal login/register page

da-vazquez committed:
SecureRouteAuth0 replaces otka secure route to ensure user is authenticated before allows routes to specific pages

da-vazquez committed:
imported Protected Route to index.js

da-vazquez committed:
domain and clientID inserted into authprovider for testing, errors with env

da-vazquez committed:
replaced okta-auth with useAuth0 for authenication verification and allows us to retrieve user info with built-in functionality

da-vazquez committed:
replaced okta authenication to auth0 isAuthenticated built-in hook. these files were changed due to rendering errors only

SethMacP reviewed (dismissed):
Great work, Daniel. You managed to get a lot done in this PR and I appreciated that you left a comment about the usefulness of that commented out code. I'd advise following up with someone outside of this PR to get an answer quicker. 

My only 'non code' critique would be to check the length of your commits. The overwhelming majority seemed to have the '...' on them.

kdneldor reviewed (commented):
Great work Daniel! You gave good details when it came to what is going on in this PR. Commits are well written and atomic. In general it seems appropriately scoped. Your motivation is clear. My main concern would be the amount of files changed, but 10 may not be a big deal.

da-vazquez committed:
removed unnecessary commented code

da-vazquez committed:
removed remaining okta components and dev comments

da-vazquez committed:
LoginContainer tests modified and passed

da-vazquez committed:
Tests updated to mock auth0 instead of okta, passed

kwnie reviewed (commented):
Looks great, Daniel. All of your code is concise and clean.

kdneldor reviewed (commented):
This PR is starting to look really good Daniel. We just need to clear up the concerns around these conversation items to get it ready for merge.

da-vazquez committed:
removed unnecessary commented code

da-vazquez committed:
consolidated imports and aligned "exact path"

da-vazquez committed:
useMemo & useEffect commented functions removed

da-vazquez committed:
updates to fix faling tests, related to userInfo prop

da-vazquez committed:
removed hardcoded domain & clientID

da-vazquez committed:
added to gitignore

tinomen reviewed (changes requested):
Lots of work to make this look simple. Well done.

kdneldor reviewed (dismissed):
This looks good Daniel! The sh.exe.stackdump file needs to be deleted, and we need to coordinate the merge with the BE, but other than that it looks ready to go.

da-vazquez committed:
Removed component as it has been replaced by NewParentDashboard

da-vazquez committed:
Removed components as they have been replaced by NewParentDashboard

da-vazquez committed:
AuthState removed and replaced with idToken and authenticated values in getAuthHeader function

da-vazquez committed:
test modified to incorporate new getIdTokenClaims hook, passed

da-vazquez committed:
idToken and isAuthenticated boolean stored in localstorage after login for streamlined access

da-vazquez committed:
getProfileData no longer requires parameter, added catch promise to console errors

da-vazquez committed:
deleted file due to switch from okta to Auth0

da-vazquez committed:
updated directory of module to RenderNewParentDashboard

da-vazquez committed:
Delete sh.exe.stackdump

Deleting file as it does not contribute anything to the project, it's just a log from my local machine

da-vazquez committed:
Merge branch 'main' into feature/auth0-login-signup

da-vazquez committed:
okta dependency must remain installed until all components migrated to auth0

da-vazquez committed:
testing updated to mock auth0 instead of okta, passed

kdneldor reviewed (approved):
Great job switching out ParentDashboard.js with NewParentDashboard.js! I'm excited to coordinate with the BE PR to get this merged!
